code_challenges/code_wars/scramble.py

```diff
@@ -0,0 +1,19 @@
+def scramble(s1, s2):
+    matching_word = unscramble(s1, s2)
+ 
+    return s2 == matching_word
+
+
+def unscramble(s1, s2):
+    matching_word = ""
+    s1 = list(s1)
+    for letter in s2:
+        if letter in s1:
+            matching_word += letter
+            s1.remove(letter)
+    return matching_word
```

Comment:
`s1 = list(s1)` - Python has to do on thing for each character in the string
`for letter in s2` - Python has to do a thing for each item in the list:
      that thing - `if letter in s1:`. How does python know if it's in the list?  It has to check each item in the list.
      also in the loop - `s1.remove(letter)` Think about the steps that have to be done by python to remove an item.  It has to first find it by checking every item until it gets there.  It then has to take it out, but what does taking it out look like?  What does it have to do to correct all the indices? 
      
Probably you can see that if the strings are very long at all you're asking python to do a whole lot of things.  How can you reduce them?